provider/cmd/pulumi-resource-xyz/main.go

```diff
@@ -12,20 +12,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:generate go run ./generate.go
 package main
 
 import (
-	_ "embed"
-	"github.com/pulumi/pulumi-xyz/provider/pkg/provider"
-	"github.com/pulumi/pulumi-xyz/provider/pkg/version"
-)
+	"math/rand"
+	"time"
 
-var providerName = "xyz"
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
 
-//go:embed schema-embed.json
-var pulumiSchema []byte
+// Version is initialized by the Go linker to contain the semver of this build.
+var Version string = "0.0.1"
 
 func main() {
-	provider.Serve(providerName, version.Version, pulumiSchema)
+	p.RunProvider("xyz", Version,
+		// We tell the provider what resources it needs to support.
+		// In this case, a single custom resource.
+		infer.Provider(infer.Options{
+			Resources: []infer.InferredResource{
+				infer.Resource[Random, RandomArgs, RandomState](),
+			},
+		}))
+}
+
+// Each resource has a controlling struct.
+type Random struct{}
```

Comment:
I think we should mention what is possible here.
```suggestion
// Resource behavior is determined by implementing methods on the controlling struct.
// The `Create` method is mandatory, but other methods are optional.
// - Check: Remap inputs before they are typed.
// - Diff: Change how instances of a resource are compared.
// - Update: Mutate a resource in place.
// - Read: Get the state of a resource from the backing provider.
// - Delete: Custom logic when the resource is deleted.
// - Annotate: Describe fields and set defaults for a resource.
// - WireDependencies: Control how outputs and secrets flows through values.
type Random struct{}
```